src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesSettings/ExternalDataSourcesSettings.js

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import arrayMutators from 'final-form-arrays';
+
+import ExternalDataSourcesView from '../ExternalDataSourcesView/ExternalDataSourcesView';
+import ExternalDataSourcesLookup from '../ExternalDataSourcesLookup/ExternalDataSourcesLookup';
+import ExternalDataSourcesFormModal from '../ExternaldataSourcesFormModal/ExternalDataSourcesFormModal';
+import ExternalDataSourceForm from '../ExternalDataSourceForm/ExternalDataSourceForm';
+
+const CREATING = 'create';
+const VIEWING = 'view';
+
+const ExternalDataSourcesSettings = ({
+  onDelete,
+  onCancel,
+  onSave,
+  externalKbs,
+  onSubmit,
+  initialValues
+}) => {
+  const [mode, setMode] = useState(VIEWING);
```

Comment:
Really nearly there.

Now we've split this into two forms, we no longer need this state here. Each form can manage it's own "open" state.

I'd prefer to have a state here called "createEDS" or something, and then a _different_ state in the view pane called "editEDS" or something like that.


That would mean that you no longer need to pass this state down at all, instead each component is responsible for managing its own "form is open" state.